### PR TITLE
`azuredevops_team` - Fixing the team admin flags

### PR DESCRIPTION
**Pull request by lukaskolafa**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Before: Team administrator was created with flags ManageMembership
After: Team administrator is created with flags Read, Write, Delete, ManageMembership, CreateScope

The behavior before is not consistent with the UI. When you create a team administrator via UI, it is consistent with the behavior after.

Issue Number: #979 

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

**Comment by lukaskolafa** [comment]
@microsoft-github-policy-service agree

**Comment by xuzhang3** [comment]
@lukaskolafa  lint check failed
```log
==> Checking that code complies with gofmt requirements...
gofmt needs running on the following files:
./azuredevops/internal/service/core/resource_team_test.go
You can use the command: `make fmt` to reformat code.
make: *** [GNUmakefile:36: fmtcheck] Error 1

```

**Comment by lukaskolafa** [comment]
thanks for the hint @xuzhang3 , I was a bit unsure with that command, because it changed 350 files in my dev environment. Anyway, I pushed the changes and it looks like it was only a whitespace change, so it should look better now.

One thing we should discuss is the backward compatibility with the old behavior. The current implementation recognizes admin by the old flag and new admins are created with the full list of flags. I believe this will work in praxis, but it is not a 100% correct implementation. What do you suggest as a resolution for this problem?

**Comment by xuzhang3** [comment]
@lukaskolafa  The difference should be caused by the line-endings characters of different platforms.(CRLF, LF).   
Not tested. Will this fix trigger an update of existing resources since permissions have changed? If so the update should help set the permission correctly.

**Comment by lukaskolafa** [comment]
@xuzhang3  I updated the code to recognize correctly admins only if all 5 flags are set. In case the resource is tainted, it will properly do an update. Unfortunately, Terraform does not recognize the difference on its own without tainting, which I believe is a standard Terraform behavior.

File: azuredevops/internal/service/core/resource_team.go [hunk 1]
```diff
@@ -563,7 +564,11 @@ func setTeamAdministratorsPermissions(d *schema.ResourceData, clients *client.Ag
 				PrincipalPermission: securityhelper.PrincipalPermission{
 					SubjectDescriptor: item.(string),
 					Permissions: map[securityhelper.ActionName]securityhelper.PermissionType{
+						"Read":             permission,
+						"Write":            permission,
+						"Delete":           permission,
 						"ManageMembership": permission,
+						"CreateScope":      permission,
```
Comment (xuzhang3) [inline azuredevops/internal/service/core/resource_team.go:570]: Checked  the default team admin and it have 6 permissions:
```
0 = Read -> 
1 = Write -> 
2 = Delete -> 
3 = ManageMembership -> 
4 = CreateScope -> 
5 = RestoreScope -> 
```

**Review by xuzhang3 — CHANGES_REQUESTED** [review]

File: azuredevops/internal/service/core/resource_team.go (hunk 1 above)
Comment (lukaskolafa) [reply azuredevops/internal/service/core/resource_team.go:570]: how exactly did you test this? I can confirm, that the `RestoreScope` bit exists, this command

`az devops security permission namespace list --organization "https://dev.azure.com/...."` gives me this result:

```
...
[
      {
        "bit": 1,
        "displayName": "View identity information",
        "name": "Read",
        "namespaceId": "5a27515b-ccd7-42c9-84f1-54c998f03866"
      },
      {
        "bit": 2,
        "displayName": "Edit identity information",
        "name": "Write",
        "namespaceId": "5a27515b-ccd7-42c9-84f1-54c998f03866"
      },
      {
        "bit": 4,
        "displayName": "Delete identity information",
        "name": "Delete",
        "namespaceId": "5a27515b-ccd7-42c9-84f1-54c998f03866"
      },
      {
        "bit": 8,
        "displayName": "Manage group membership",
        "name": "ManageMembership",
        "namespaceId": "5a27515b-ccd7-42c9-84f1-54c998f03866"
      },
      {
        "bit": 16,
        "displayName": "Create identity scopes",
        "name": "CreateScope",
        "namespaceId": "5a27515b-ccd7-42c9-84f1-54c998f03866"
      },
      {
        "bit": 32,
        "displayName": "Restore identity scopes",
        "name": "RestoreScope",
        "namespaceId": "5a27515b-ccd7-42c9-84f1-54c998f03866"
      }
    ]
...
```

but, when I create a new Project and check the permissions using this command (PowerShell):

`$namespacePermissions = Invoke-RestMethod -Uri "$organizationUri/_apis/accesscontrollists/$($namespace.namespaceId)?api-version=7.1-preview.1" -Method GET -Headers $authenticationHeader`

Then I retrieve the value `allow = 31` for the default Team administrator, which seems to be the value stored in AZDO DB.

Comment (xuzhang3) [reply azuredevops/internal/service/core/resource_team.go:570]: You are right, the admin allow value is 31.

Comment (xuzhang3) [reply azuredevops/internal/service/core/resource_team.go:570]: Lint check failed:
```log
azuredevops/internal/service/core/resource_team.go:498:44: unnecessary leading newline (whitespace)
```

Comment (lukaskolafa) [reply azuredevops/internal/service/core/resource_team.go:570]: sorry, I fixed the lines

**Review by xuzhang3 — APPROVED** [review]

**Pull request merged**